src/services/refactors/extractSymbol.ts

```diff
@@ -295,6 +300,10 @@ namespace ts.refactor.extractSymbol {
             return { errors: [createFileDiagnostic(sourceFile, span.start, length, Messages.cannotExtractRange)] };
         }
 
+        if (isJSDoc(start)) {

```

Comment:
Why did this move?  Was it an optimization or did it somehow affect correctness?

Reply:
When a JSDoc comment is selected, `findTokenOnLeftOfPosition` gives the `}` token in the JSDoc type tag, so snapping to the endToken doesn't match the end of the JSDoc comment. However, there is no meaningful extraction for a JSDoc comment so this would fail with a generic `cannotExtractRange` message, and we only check the `start` node for this error anyway, so I just moved it up so we continue to offer the more meaningful error message.